libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -41,16 +64,15 @@ bool IRGeneratorForStatements::visit(VariableDeclarationStatement const& _varDec
 
 		expression->accept(*this);
 
-		solUnimplementedAssert(
-			*expression->annotation().type == *_varDeclStatement.declarations().front()->type(),
-			"Type conversion not yet implemented"
-		);
 		m_code <<
 			"let " <<
 			m_context.variableName(*_varDeclStatement.declarations().front()) <<
-			" := " <<
-			m_context.variable(*expression) <<
-			"\n";
+			" := ";
+
+		Type const& from = *_varDeclStatement.declarations().front()->type();
+		Type const& to   = *expression->annotation().type;
+
+		appendConvertIfRequired(from, to, m_context.variable(*expression));
```

Comment:
I think it is nicer to return Yul code as string in all cases except for the visitors.